### PR TITLE
Ignore `__MACOSX` folder in tool archives when launching core install

### PR DESCRIPTION
**Pull request by alessio-perugini**
## Please check if the PR fulfills these requirements

See [how to contribute](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/)

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [x] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] `UPGRADING.md` has been updated with a migration guide (for breaking changes)
- [ ] `configuration.schema.json` updated if new parameters are added.

## What kind of change does this PR introduce?

This ignore `__MACOSX` autogenerated folder inside the archives. When we launch the `core install` command
if the downloaded archive, besides the root dir, contains also the __MACOSX we simply ignore that dir.

## What is the current behavior?

If we encounter a `__MACOSX` folder we throw an error.

## What is the new behavior?

We ignore every `__MACOSX` folder when searching for the package root.

## Does this PR introduce a breaking change, and is [titled accordingly](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#breaking)?

nope

## Other information

I've added a commit to propose a small refactoring feel free to revert it.

### CODE REVIEW

File: arduino/resources/install_test.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
+package resources
```
Comment (cmaglie) [inline arduino/resources/install_test.go:1]: Missing license headers

**Review by cmaglie — CHANGES_REQUESTED** [review]
LGTM! just a missing license header.

**Comment by cmaglie** [comment]
We should also update the documentation and explicitly say that in the archives the subdirectory `__MACOSX` is ignored.

BTW I can't really find a place where the archive format is specified, the only place where this topic is discussed is at the bottom of `docs/package_index_json-specification.md`:

> ## Installation archive structure
>
> The installation archives contain the Board support files.
>
> Supported formats are `.zip`, `.tar.bz2`, and `.tar.gz`. Starting from Arduino CLI >=0.30.0 support for `.tar.xz`, and
> `.tar.zst` has been added, by the way, if you want to keep compatiblity with older versions of Arduino IDE and Arduino
> CLI we recommend using one of the older formats.
>
> The folder structure of the core archive is slightly different from the standard manually installed Arduino IDE 1.5+
> compatible hardware folder structure. You must remove the architecture folder(e.g., `avr` or `arm`), moving all the
> files and folders within the architecture folder up a level.

the hint for "The folder structure of the core archive" is referred specifically for platform archives. So... I guess this documentation requires a bit of refactoring, maybe it's worth opening another issue.

**Review by cmaglie — APPROVED** [review]

**Comment by alessio-perugini** [comment]
> We should also update the documentation and explicitly say that in the archives the subdirectory `__MACOSX` is ignored.
> 
> BTW I can't really find a place where the archive format is specified, the only place where this topic is discussed is at the bottom of `docs/package_index_json-specification.md`:
> 
> > ## Installation archive structure
> > The installation archives contain the Board support files.
> > Supported formats are `.zip`, `.tar.bz2`, and `.tar.gz`. Starting from Arduino CLI >=0.30.0 support for `.tar.xz`, and
> > `.tar.zst` has been added, by the way, if you want to keep compatiblity with older versions of Arduino IDE and Arduino
> > CLI we recommend using one of the older formats.
> > The folder structure of the core archive is slightly different from the standard manually installed Arduino IDE 1.5+
> > compatible hardware folder structure. You must remove the architecture folder(e.g., `avr` or `arm`), moving all the
> > files and folders within the architecture folder up a level.
> 
> the hint for "The folder structure of the core archive" is referred specifically for platform archives. So... I guess this documentation requires a bit of refactoring, maybe it's worth opening another issue.

We'll be taking care to improve docs in a separate PR. Tracking on #2167 :rocket:

**Pull request merged**